app/webpack.config.js

```diff
@@ -296,29 +303,35 @@ module.exports = [
         // just keep the woff2 fonts from fontawesome
         {
           test: /fontawesome-free.*\.(svg|eot|ttf|woff)$/,
-          loader: 'ignore-loader',
+          exclude: /fontawesome-free.*\.woff2$/,
         },
         {
           test: /\.(jpe?g|png|gif|ico|eot|ttf|map|woff2?)(\?v=\d+\.\d+\.\d+)?$/i,
           type: 'asset/resource',
         },
         {
-          test: /\.js$/,
-          use: ['source-map-loader'],
+          resourceQuery: /text/,
+          type: 'asset/resource',
+          generator: {
+            filename: '[name]-[contenthash:7][ext]',
+          },
         },
       ],
     },
     optimization: {
       moduleIds: 'deterministic',
+      splitChunks: {
+        chunks: 'all',
+        cacheGroups: {
+          jlab_core: {
+            test: /[\\/]node_modules[\\/]@(jupyterlab|jupyter\-notebook|lumino(?!\/datagrid))[\\/]/,
```

Comment:
what's the hit like for including the `jupyter-notebook` assets in the core bundle for non-notebook apps?

Reply:
Good question. We could also leave it out for now and investigate more later.